react/src/components/Files/FileListing.tsx

```diff
@@ -65,20 +67,28 @@ export const FileListing: React.FC<FileListingProps> = ({
 
   const { data: user } = useAuthenticatedUser();
 
-  const [chonkyFiles, setChonkyFiles] = React.useState<any>([]);
+  const [chonkyFiles, setChonkyFiles] = React.useState<any>(
+    new Array(8).fill(null)
+  );
   const [folderChain, setFolderChain] = React.useState<any>([]);
-  const [path, setPath] = React.useState<string>('');
   const [selectedSystem, setSelectedSystem] = React.useState<TTapisSystem>(
     myDataSystem as TTapisSystem
   );
   const [hasError, setHasError] = React.useState(false);
+  const [loadingMoreFiles, setLoadingMoreFiles] = React.useState(false);
+  const [hasMoreFiles, setHasMoreFiles] = React.useState(false);
+  const [listingState, setListingState] = React.useState({
+    path: '',
+    offset: 0,
+  });
 
   const { data: files, refetch } = useFiles({
     system: selectedSystem?.id || '',
     path:
-      path || (selectedSystem?.id === myDataSystem?.id ? user?.username : ''),
-    offset: '0',
-    limit: '100' /* TODO https://tacc-main.atlassian.net/browse/WG-418 */,
+      listingState.path ||
+      (selectedSystem?.id === myDataSystem?.id ? user?.username : ''),
+    offset: listingState.offset.toString(),
+    limit: DEFAULT_FILE_LIMIT.toString(),
```

Comment:
Few suggestions for future improvement:
- offset and limit should be type int, and adjust in the hook query params
- react query has a nice [infinite query hook](https://tanstack.com/query/latest/docs/framework/react/guides/infinite-queries). Used in tandem with `InterSectionObserver`, or `react-intersection-observer`, this would simplify this code a bit with the use of a ref for a node at the bottom of the listing, and a callback to the react-query built-in `fetchNextPage()` method. If gone that route, you'd be able to get rid of most of the `useState`s in this component